x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/user/TransportSetEnabledAction.java

```diff
@@ -51,9 +50,6 @@ protected void doExecute(Task task, SetEnabledRequest request, ActionListener<Ac
         if (securityContext.getUser().principal().equals(request.username())) {
             listener.onFailure(new IllegalArgumentException("users may not update the enabled status of their own account"));
             return;
```

Comment:
uff nice catch! Will address in a follow up PR 👍

Reply:
> Also, since we prevent user from disabling itself, why should we allow it delete itself (TransportDeleteUserAction)?

Agreed that this is inconsistent. Two options: remove the restriction in set enabled or add it to `TransportDeleteUserAction`. Former isn't ideal since it seems like a sensible safeguard (plus prevents users from re-enabling themselves). Latter makes sense but would be a breaking change. Not sure if it's worth the hassle tbh.

Reply:
Let's be cautions about changing the behaviour of deleting users. I was merely raising the question, not a proposal. But the set enabled user behaviour is a bug and should be fixed (by checking realm alongside username). Btw, users cannot re-enable themseleves even without this check. Disabled users fail at authentication time